packages/shared/src/authz/fragments/workspaces.ts

```diff
@@ -24,47 +27,58 @@ export const maybeMemberRoleWithValidSsoSessionIfNeeded: AuthPolicyFragment<
     // Get workspace, so we can resolve its slug for error scenarios
     const workspace = await loaders.getWorkspace({ workspaceId })
     // hides the fact, that the workspace does not exist
-    if (!workspace) return just(err(new WorkspaceNoAccessError()))
+    if (!workspace) return err(new WorkspaceNoAccessError())
 
     const hasAnyRole = await hasAnyWorkspaceRole(loaders)({ userId, workspaceId })
-    if (!hasAnyRole) return just(err(new WorkspaceNoAccessError()))
+    if (!hasAnyRole) return err(new WorkspaceNoAccessError())
 
     const hasMinimumMemberRole = await requireMinimumWorkspaceRole(loaders)({
       userId,
       workspaceId,
       role: 'workspace:member'
     })
     // only members and above need to use sso
-    if (!hasMinimumMemberRole) return nothing()
+    if (!hasMinimumMemberRole) return ok()
```

Comment:
no more `Maybe` results -a fragment only makes a decision for itself, not for the parent policy. it either succeeds or it fails, so no `maybe` necessary

Reply:
Ok, with the implicit project role moved into where it is now, this can work.